packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart

```diff
@@ -137,6 +157,17 @@ class PurchaseDetails {
   /// This is null on Android.
   final PurchaseWrapper billingClientPurchase;
 
+  /// The developer has to call [InAppPurchaseConnection.completePurchase] if the value is `true`.
+  ///
+  /// The initial value is `false`.
+  /// * See also [InAppPurchaseConnection.completePurchase] for more details on completing purchases.
+  bool pendingCompletePurchase = false;
```

Comment:
This is editable by anything. It seems like it should probably be split up into a getter and setter around a private variable so it can only be manipulated internally, does that make sense?

Reply:
done